src/Analyzers/CSharp/Tests/UseParameterNullChecking/UseParameterNullCheckingTests.cs

```diff
@@ -0,0 +1,981 @@
+    {
+        if (c == null)
+            throw new ArgumentNullException(nameof(c));
+    }
+}
+";
+            await new VerifyCS.Test()
+            {
+                TestCode = testCode,
+                FixedCode = testCode,
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestWithUnusedUserDefinedOperator()
+        {
+            await new VerifyCS.Test()
+            {
+                TestCode = @"using System;
+class C
+{
+    public static bool operator ==(C c1, C c2) => true;
+    public static bool operator !=(C c1, C c2) => false;
+
+    static void M(C c)
+    {
+        [|if ((object)c == null)
+            throw new ArgumentNullException(nameof(c));|]
+    }
+}
+",
+                FixedCode = @"using System;
+class C
+{
+    public static bool operator ==(C c1, C c2) => true;
+    public static bool operator !=(C c1, C c2) => false;
+
+    static void M(C c!!)
+    {
+    }
+}
+",
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestWithPreprocessorDirective()
+        {
+            await new VerifyCS.Test()
+            {
+                TestCode = @"using System;
+class C
+{
+    static void M(C c)
+    {
+#nullable enable
+        [|if ((object)c == null)
+            throw new ArgumentNullException(nameof(c));|]
+#nullable disable
+    }
+}
+",
+                FixedCode = @"using System;
+class C
+{
+    static void M(C c!!)
+    {
+#nullable disable
+    }
+}
+",
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestPartialMethod()
+        {
+            var partialDeclaration = @"
+partial class C
+{
+    static partial void M(C c);
+}
+";
+
+            await new VerifyCS.Test()
+            {
+                TestState =
+                {
+                    Sources =
+                    {
+                        partialDeclaration,
+                        @"using System;
+partial class C
+{
+    static partial void M(C c)
+    {
+        [|if ((object)c == null)
+            throw new ArgumentNullException(nameof(c));|]
+    }
+}
+"
+                    }
+                },
+                FixedState =
+                {
+                    Sources =
+                    {
+                        partialDeclaration,
+                        @"using System;
+partial class C
+{
+    static partial void M(C c!!)
+    {
+    }
+}
+"
+                    }
+                },
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
```

Comment:
can you add tests for all other types the compiler supports putting `!!` on (and all other types the compiler would *error* on putting !! on).  For example, if !! is allowed on pointers, can we add a test for that?

i may have missed it, but a test of just a simple value type would be good.  as well as a tests with a nullable value type.

Reply:
I'll make another test pass with this in mind.